### PR TITLE
Add support for attachments

### PR DESCRIPTION
**Pull request by christopherdro**

### CODE REVIEW

File: RNMail/RNMail.m [hunk 1]
```diff
@@ -47,6 +47,39 @@ - (dispatch_queue_t)methodQueue
...
+            if (!attachmentName) {
+                attachmentName = [[attachmentPath lastPathComponent] stringByDeletingPathExtension];
+            }
+            
+            // Get the resource path and read the file using NSData
+            NSData *fileData = [NSData dataWithContentsOfFile:attachmentPath];
+            
+            // Determine the MIME type
+            NSString *mimeType;
+            if ([attachmentType isEqualToString:@"jpg"]) {
```
Comment (chirag04) [inline RNMail/RNMail.m]: would a switch case be easy to maintain?

File: README.md [hunk 2]
```diff
@@ -21,7 +22,10 @@ var MailExampleApp = React.createClass({
     Mailer.mail({
       subject: 'need help',
       recipients: ['support@example.com'],
-      body: ''
+      body: '',
+      attachmentPath: '',  // The absolute path of the file from which to read data.
+      attachmentType: '',   // Mime Type: jpg, png, doc, ppt, html, pdf
+      attachmentName: '',   // Optional: Custom filename for attachment
```
Comment (chirag04) [inline README.md]: can we make one object with these keys instead?

```
attachment: {
   path
   type
   name
}
```

Comment (christopherdro) [inline README.md]: Updated with latest commit

File: RNMail/RNMail.m (hunk 1 above)
Comment (christopherdro) [inline RNMail/RNMail.m]: Switch statements don´t work with NSStrings, only with integers.

File: RNMail/RNMail.m [hunk 3]
```diff
@@ -47,6 +47,40 @@ - (dispatch_queue_t)methodQueue
             [mail setToRecipients:recipients];
         }
         
+        if (options[@"attachment"][@"path"] && options[@"attachment"][@"name"]){
```
Comment (chirag04) [inline RNMail/RNMail.m]: bail out early. check if attachment is available first and then attachment[path].

Also, If name is optional then why check for it here? Did you mean type?

**Comment by chirag04** [comment]
Awesome. one last nit i guess and then will merge it.

File: RNMail/RNMail.m (hunk 3 above)
Comment (christopherdro) [inline RNMail/RNMail.m]: Yes, sorry for the typo. Should be fixed

**Comment by chirag04** [comment]
:+1: Thanks @christopherdro

**Comment by christopherdro** [comment]
:+1: np!

**Pull request merged**